state/protocol/badger/mutator_test.go

```diff
@@ -1653,6 +1655,60 @@ func TestHeaderExtendHighestSeal(t *testing.T) {
 	})
 }
 
+// TestExtendInvalidGuarantee checks if Extend method will reject invalid blocks that contain
+// guarantees with invalid guarantors
+func TestExtendInvalidGuarantee(t *testing.T) {
```

Comment:
## 🔍 (requesting to test an additional scenarios)

very nice test. Great documentation. Would be great to explicitly test the following scenarios: 
* `guarantee.ChainID` (identifying the collector cluster) has an unknown value? 
* illegal patting 
* invalid checksum

I think a [similar comment](https://github.com/onflow/flow-go/pull/2414#discussion_r876732747) here has lead to some confusion: 
* the purpose of these tests is _not_ to verify that `FindGuarantors` returns the correct sentinel error 
* the purpose is to test that `MutableState` correctly handles the error:
   * for _expected_ sentinel errors, `MutableState` wraps them as `InvalidExtensionError` 
   * `MutableState` does _not_ misinterpret unexpected exceptions as the input being invalid


unfortunately, we have to test these scenarios over and over again, because we can't mock the logic in `FindGuarantors`. I think adding a validator interface for collection guarantees might be valuable. The validator implementation could then call into `FindGuarantors`. Thereby, we can independently test the logic of detecting the byzantine cases (in validator), and correct handling of the expected lower-level sentinels (`MutableState`)

Reply:
Yeah, that's right. In the unit tests, I'm able to let FindGuarantees to return all the sentinel errors except `ErrEpochNotCommitted` error. That error isn't easy to create without creating a `guaranteeValidator` just like the `sealValidator` and `receiptValidator`. However, adding `guaranteeValidator` would create a lot of code changes. Will be addressed in a separate PR.